src/bin/cli/mod.rs

```diff
@@ -0,0 +1,1052 @@
+            self
+                ._arg(
+                    opt("features", "Space-separated list of features to activate")
+                        .value_name("FEATURES")
+                )
+                ._arg(opt("all-features", "Activate all available features"))
+                ._arg(opt("no-default-features", "Do not activate the `default` feature"))
+        }
+
+        fn arg_release(self, release: &'static str) -> Self {
+            self._arg(opt("release", release))
+        }
+
+        fn arg_target_triple(self, target: &'static str) -> Self {
+            self._arg(opt("target", target).value_name("TRIPLE"))
+        }
+
+        fn arg_manifest_path(self) -> Self {
+            self._arg(opt("manifest-path", "Path to Cargo.toml").value_name("PATH"))
+        }
+
+        fn arg_message_format(self) -> Self {
+            self._arg(
+                opt("message-format", "Error format")
+                    .value_name("FMT")
+                    .case_insensitive(true)
+                    .possible_values(&["human", "json"]).default_value("human")
+            )
+        }
+
+        fn arg_new_opts(self) -> Self {
+            self._arg(
+                opt("vcs", "\
+Initialize a new repository for the given version \
+control system (git, hg, pijul, or fossil) or do not \
+initialize any version control at all (none), overriding \
+a global configuration.")
+                    .value_name("VCS")
+                    .possible_values(&["git", "hg", "pijul", "fossil", "none"])
+            )
+                ._arg(opt("bin", "Use a binary (application) template [default]"))
+                ._arg(opt("lib", "Use a library template"))
+                ._arg(
+                    opt("name", "Set the resulting package name, defaults to the directory name")
+                        .value_name("NAME")
+                )
+        }
+
+        fn arg_index(self) -> Self {
+            self
+                ._arg(
+                    opt("index", "Registry index to upload the package to")
+                        .value_name("INDEX")
+                )
+                ._arg(
+                    opt("host", "DEPRECATED, renamed to '--index'")
+                        .value_name("HOST")
+                        .hidden(true)
+                )
+        }
+    }
+
+    impl CommonArgs for App {
+        fn _arg(self, arg: Arg<'static, 'static>) -> Self {
+            self.arg(arg)
+        }
+    }
+
+    pub fn opt(name: &'static str, help: &'static str) -> Arg<'static, 'static> {
+        Arg::with_name(name).long(name).help(help)
+    }
+
+    pub fn subcommand(name: &'static str) -> App {
+        SubCommand::with_name(name)
+            .settings(&[
+                AppSettings::UnifiedHelpMessage,
+                AppSettings::DeriveDisplayOrder,
+                AppSettings::DontCollapseArgsInUsage,
+            ])
+    }
+}
+
+
+
+fn values<'a>(args: &ArgMatches, name: &str) -> &'a [String] {
```

Comment:
Obviously, this needs to be fixed properly :) Docopt gives us `Vec<String>`, and `clap` `Iterator<Item=&str>`, and we unfortunately use `&'a [String]` internally....